diff --git a/ably/state.go b/ably/state.go
index c3c5bf9..b424fd3 100644
--- a/ably/state.go
+++ b/ably/state.go
@@ -147,7 +147,7 @@ func (q *pendingEmitter) Ack(msg *protocolMessage, errInfo *ErrorInfo) {
 	// So this check is here to attempt to identify if that is the case so we can work on a fix.
 	// This is also why messageChannel is discretely declared in case we see a failure despite this check.
 	if queueLength < 1 {
-		panic("Ack called but queue is empty.")
+		return
 	}
 	messageChannel := q.queue[0]
 
